anomalib/deploy/export.py

```diff
@@ -50,39 +52,77 @@ def get_model_metadata(model: AnomalyModule) -> dict[str, Tensor]:
     return meta_data
 
 
+def get_metadata_from_trainer(trainer: pl.Trainer) -> dict[str, Any]:
+    """Get meta data information from PL Trainer.
+
+    Args:
+        trainer (pl.Trainer): Pytorch Lightning trainer object containing model, datamodule and the rest of the hparams.
+
+    Returns:
+        dict[str, Any]: metadata
+    """
+    model = trainer.model
+
+    metadata: dict[str, Any] = {
+        "task": trainer.datamodule.test_data.task,
+        "transform": trainer.datamodule.test_data.transform.to_dict(),
+        "image_threshold": model.image_threshold.cpu().value.item(),
+        "pixel_threshold": model.pixel_threshold.cpu().value.item(),
+    }
+
+    # Add normalization metrics to the post processing metadata
+    if hasattr(model, "normalization_metrics") and model.normalization_metrics.state_dict() is not None:
+        for key, value in model.normalization_metrics.state_dict().items():
+            metadata[key] = value.cpu()
+            metadata[key] = value.cpu()
+
+    return metadata
+
+
 def export(
-    model: AnomalyModule,
-    input_size: list[int] | tuple[int, int],
-    export_mode: ExportMode,
-    export_root: str | Path,
+    trainer: pl.Trainer,
```

Comment:
The only extra thing we get from trainer is the task and transform. Can't we pass this information explicitly to this method. We can reuse this for the CLI.